File: Data Structures & Algorithms/search-2d-matrix/submission-5.java
class Solution {
    public boolean searchMatrix(int[][] matrix, int target) {
        int rowMAX =  matrix.length;
        int colMAX =  matrix[0].length;

        int low = 0;
        int high = rowMAX*colMAX-1;
        while(low <= high){
            int mid = low + (high-low)/2;
            int row = mid / colMAX;
            int col = mid % colMAX;

            if(matrix[row][col] == target){
                return true;
            }
            else if(matrix[row][col] > target){
                high = mid-1;
            }
            else{
                low = mid+1;
            }
        }


        return false;
    }
}

File: Data Structures & Algorithms/binary-search/submission-5.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        def searchRec( nums, target, low, high):
             if low > high:
                return -1
            
             mid = low + (high-low)//2
             if(nums[mid] == target):
                return mid
             elif(nums[mid] < target):
                return searchRec(nums, target, mid+1, high)
             else:
                return searchRec(nums, target, low, mid-1)
        
        return searchRec(nums, target, 0, len(nums)-1)

       

        